card/db/migrate_core_cards/20150429090551_search_card_context.rb

```diff
@@ -0,0 +1,29 @@
+# -*- encoding : utf-8 -*-
+
+class SearchCardContext < Card::CoreMigration
+  def up
+    sep = %r{\W}
+    replace = [
+      ['[lr]+','l\\1'],
+      ['[LR]+','L\\1'],
+      ['(?=[LR]*[lr]+)(?=[lr]*[LR]+)[lrLR]+','l\\1'],   # mix of lowercase and uppercase l's and r's
+      ['left',   'LL'],
+      ['right',  'LR'],
+      ['self',   'left'],
+      ['',       'left'],
+    ]
+    Card.search(:type_id=>Card::SearchTypeID).each do |card|
+      if card.cardname.junction?
+        content = card.content
+        replace.each do |key, val|
+          content.gsub!(/(?<=#{sep})_(#{key})(?=#{sep})/, "_#{val}")
+        end
+        card.update_attributes! :content=>content
+      end
+    end
+    Card["*self+*right+*structure"].update_attributes! :content=>'{"name":"_left"}'
+    Card["*right+*right+*structure"].update_attributes! :content=>'{"right":"_left"}'
+    Card["*type+*right+*structure"].update_attributes! :content=>'{"type":"_left"}'
+    Card["*type plus right+*right+*structure"].update_attributes! :content=>'{"left":{"type":"_LL"}, "right":"_LR"}'
```

Comment:
Re #3, now that I think about it, yes and (I hope) yes.

Any Set with a (card) rule has an actual set card associated with it, eg http://wikirate.org/Company+*type?view=raw, where the db_content is `{"type":"_self"}`.  That would, of course, be incorrect WQL for that set with the new syntax.

I *think* it should be safe, because the WQL in those cards is basically used like any WQL in a search context, eg http://wikirate.org/Company+*type+by_name

At any rate, I do suspect it's unsafe *not* to do it, because use the above pattern quite a lot.